src/execute.rs

```diff
@@ -426,10 +426,346 @@ impl Execute for System {
+        self.registers.a = (value >> 1) | (self.registers.sr.read(flags::CARRY) as u8) << 7;
+        self.registers.alu_add(value);
+
+        self.registers.sr.write(flags::CARRY, value & 0x01 != 0);
+        self.registers.sr.set_nz(self.registers.a);
+        Ok(2)
+      }
+
+      0x63 | 0x67 | 0x73 | 0x77 | 0x7B | 0x7F => {
+        // RRA: ROR -> ADC
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value >> 1 | (self.registers.sr.read(flags::CARRY) as u8) << 7;
+
+        self.registers.sr.write(flags::CARRY, value & 0x01 != 0);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        self.registers.alu_add(result);
+
+        Ok(cycles)
+      }
+
+      0x83 | 0x87 | 0x8F | 0x97 => {
+        // SAX: AND -> STA
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.registers.x & self.registers.a;
+        self.registers.sr.set_nz(value);
+        self.write(address, value);
+
+        Ok(cycles)
+      }
+
+      0xA3 | 0xA7 | 0xAF | 0xB3 | 0xB7 | 0xBF => {
+        // LAX: LDA & LDX
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        self.registers.a = value;
+        self.registers.x = value;
+        self.registers.sr.set_nz(value);
+
+        Ok(cycles)
+      }
+
+      0xC3 | 0xC7 | 0xCF | 0xD3 | 0xD7 | 0xDB | 0xDF => {
+        // DCP: DEC + SEC
+        self.registers.sr.set(flags::CARRY);
+
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value.wrapping_sub(1);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0xE3 | 0xE7 | 0xEF | 0xF3 | 0xF7 | 0xFB | 0xFF => {
+        // ISC: INC => SBC
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value.wrapping_add(1);
+        self.registers.alu_subtract(value);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0x0B | 0x2B => {
+        // ANC: AND byte with accumulator. If result is negative then carry is set.
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = self.registers.a & value;
+        self.registers.sr.write(flags::CARRY, new_val & 0x80 != 0);
+
+        Ok(cycles)
+      }
+
+      0x4B => {
+        // ALR: AND + LSR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = (self.registers.a & value) >> 1;
+
+        self.registers.sr.write(flags::CARRY, new_val & 0x01 != 0);
+        self.registers.sr.set_nz(new_val);
+
+        Ok(cycles)
+      }
+
+      0x6B => {
+        // ARR: AND + ROR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let mut new_val = self.registers.a & value;
```

Comment:
```suggestion
        let new_val = self.registers.a & value;
```

variable does not need to be mutable, since the `let` binding further down just creates a new variable with the same name